main/src/test/java/org/mobilitydata/gtfsvalidator/validator/ShapeUsageValidatorTest.java

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright 2020 Google LLC, MobilityData IO
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.mobilitydata.gtfsvalidator.validator;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.mobilitydata.gtfsvalidator.notice.NoticeContainer;
+import org.mobilitydata.gtfsvalidator.notice.UnusedShapeNotice;
+import org.mobilitydata.gtfsvalidator.table.GtfsShape;
+import org.mobilitydata.gtfsvalidator.table.GtfsShapeTableContainer;
+import org.mobilitydata.gtfsvalidator.table.GtfsTrip;
+import org.mobilitydata.gtfsvalidator.table.GtfsTripTableContainer;
+import org.mockito.ArgumentCaptor;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static com.google.common.truth.Truth.assertThat;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.verifyNoInteractions;
+import static org.mockito.Mockito.verifyNoMoreInteractions;
+import static org.mockito.Mockito.when;
+
+public class ShapeUsageValidatorTest {
```

Comment:
This is another rule that I believe, to my knowledge, the spec doesn't explicitly enforce (all shapes in shapes.txt must be referenced in trips.txt). But it's a good idea to remove unused shapes from a GTFS file to reduce the file size and processing by consumers. So maybe warning instead of error? It depends how we want to define warnings.

@timMillet @scmcca Thoughts?

Reply:
Similar issue https://github.com/MobilityData/gtfs-validator/issues/417

Reply:
I agree with a warning. All unnecessary information should not lead to an error, as the dataset and all the information contained in it may still be valid. 
This situation should have the same impact as when producers are providing extra fields or extra files.